### PR TITLE
BLD: Move build to setuptools

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @willow-ahrens @hameerabbasi,

This PR updates `pyproject.toml` to match `pydata/sparse` build setup.

### CODE REVIEW

File: finch/_version.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# file generated by setuptools_scm
```
Comment (hameerabbasi) [inline finch/_version.py]: This file shouldn't be in version control and should be added to `.gitignore`.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
LGTM, just the one minor change.

File: finch/_version.py (hunk 1 above)
Comment (mtsokol) [reply finch/_version.py]: Ah right - done!

**Comment by hameerabbasi** [comment]
I think the readme needs updating, I'd also search for other mentions of Poetry.

**Comment by mtsokol** [comment]
> I think the readme needs updating, I'd also search for other mentions of Poetry.

Sure! I updated `README.md` and `.gitignore` to remove all occurrences of Poetry.

**Review by hameerabbasi — APPROVED** [review]

**Comment by willow-ahrens** [comment]
I don't have much of value to add to this. Using setuptools seems really important to y'all, so I'm happy to go along with it, especially since it seems like setuptools might support an opportunity to add a build step. LMK if there's anything I need to do (i.e. adding pypi keys to github secrets, etc.) to get the next version published.

**Pull request merged**